packages/databases-collections/src/stores/collections-store.js

```diff
@@ -13,18 +13,25 @@ import { collectionsReducer } from '../modules';
 const store = createStore(collectionsReducer, applyMiddleware(thunk));
 
 store.onActivated = (appRegistry) => {
-  const onCollectionsChange = throttle(
-    (collections, databases, dbName = store.getState().databaseName) => {
-      collections = collections ?? databases.get(dbName)?.collections;
-      if (collections && collections.parent.getId() === dbName) {
+  const onCollectionsChange = throttle((collections, force = false) => {
```

Comment:
how does this work? are simultaneous invocations discarded or enqueued? Like could it be that we miss an update here?

Reply:
First one and the last one always fires, everything in between is throttled by dropping them, but firing the closest one to the throttled time span (so like every 100ms one will fire). We will miss some updates, but will get the first and the last state which is the most important ones. The ones we loose are also not that important here (nor in the databases-store), they will be intermediate dbStats / collStats populating data in separate database / collection instances.

Generally speaking I would prefer us not to have any throttling but it helps to mitigate the issues with React reconciling too much when state updates happen. Does this sounds alright to you?

Reply:
Sounds great yeah! thanks for the explanation